### PR TITLE
First pass at `ste_vec` docs for JSONB containment indexing

### PR DESCRIPTION
**Pull request by freshtonic**
It's wordier than the doc for the other index types. I might have gone into too much detail - either that, or the other index types need more detail. Unsure.

@coderdan can you fact check my explanation of the purpose of the `prefix` (salt-like thingo) please?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,6 +207,100 @@ If you're using n-gram as a token filter, then a token that is already shorter t
 However, if that same short string only appears as a part of a larger token, then it will not match that record.
 In general, therefore, you should try to ensure that the string you search for is at least as long as the `tokenLength` of the index, except in the specific case where you know that there are shorter tokens to match, _and_ you are explicitly OK with not returning records that have that short string as part of a larger token.
 
+###### ste_vec opts
+
+An ste_vec index on a encrypted JSONB column enables the use of Postgres's `@>` and `<@` containment operators.
```
Comment (auxesis) [inline README.md:212]: Link to the Postgres documentation so users have more context
```suggestion
An ste_vec index on a encrypted JSONB column enables the use of PostgreSQL's `@>` and `<@` [containment operators](https://www.postgresql.org/docs/16/functions-json.html#FUNCTIONS-JSONB-OP-TABLE).
```

File: README.md [hunk 2]
```diff
@@ -205,6 +207,100 @@ If you're using n-gram as a token filter, then a token that is already shorter t
...
+
+An ste_vec index requires one piece of configuration: the `prefix` (a string) which is functionally similar to a salt for the hashing process.
+
+Within a dataset, encrypted columns indexed using an ste_vec that use different prefixes can never compare as equal and containment queries that manage to mix index terms from multiple columns will never return a positive result. This is by design.
```
Comment (auxesis) [inline README.md:216]: Line per sentence, to improve editability
```suggestion
Within a dataset, encrypted columns indexed using an ste_vec that use different prefixes can never compare as equal. 
Containment queries that manage to mix index terms from multiple columns will never return a positive result. 
This is by design.
```

File: README.md [hunk 3]
```diff
@@ -205,6 +207,100 @@ If you're using n-gram as a token filter, then a token that is already shorter t
...
+
+The index is generated from a JSONB document by first flattening the structure of the document such that a hash can be generated for each unique path prefix to a node.
+
+The complete set of JSON types is supported by the indexer. Null values are ignored by the indexer.
```
Comment (auxesis) [inline README.md:220]: Line per sentence, to improve editability

```suggestion
The complete set of JSON types is supported by the indexer. 
Null values are ignored by the indexer.
```

**Review by auxesis — CHANGES_REQUESTED** [review]
This is great to see @freshtonic. 

Have added some minor suggestions.

File: README.md [hunk 4]
```diff
@@ -209,7 +209,7 @@ In general, therefore, you should try to ensure that the string you search for i
 
 ###### ste_vec opts
 
-An ste_vec index on a encrypted JSONB column enables the use of Postgres's `@>` and `<@` containment operators.
+An ste_vec index on a encrypted JSONB column enables the use of PostgreSQL's `@>` and `<@` [containment operators](https://www.postgresql.org/docs/16/functions-json.html#FUNCTIONS-JSONB-OP-TABLE).
```
Comment (freshtonic) [inline README.md:319]: Nice

**Comment by freshtonic** [comment]
Looks like @calvinbrewer has actioned your suggestions already, @auxesis .

Let me know when this is good to merge.

**Review by auxesis — APPROVED** [review]
@freshtonic LGTM, :shipit:

**Pull request merged**